### PR TITLE
Updates for retrofit in gradle, updates for EngineOil EngineOilTemperature models

### PR DESCRIPTION
**Pull request by oleksiiborys**
Gradle update
Removed Jacoco
Updated retrofit version
Updated EngineOil and EngineOilTemperature models

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -27,37 +26,3 @@ allprojects {
...
-    onlyIf = {
-        true
-    }
-
-    doFirst {
-        executionData = files(executionData.findAll {
-            it.exists()
-        })
-    }
-}
```
Comment (JesseScott) [inline build.gradle]: I know that JaCoCo can cause some compile time lags, as it Instruments and pre-dexes the code...
But i'd prefer to just _disable it_ and then have to _enable it_ via a Flag...

Eg.
```
    buildTypes {
        debug {
            testCoverageEnabled false
        }
        release {
            // https://jeroenmols.com/blog/2016/09/01/coveragecost/
            testCoverageEnabled jacocoEnabled.toBoolean()
       }
    }
```
which you would invoke by doing 
```
./gradlew -PjacocoEnabled=true clean test jacocoRootReport
```
... or something to that effect.
See https://jeroenmols.com/blog/2016/09/01/coveragecost/

A really cool thing that JaCoCo can do that I recently learned is that you can use it to enforce Code Coverage
---> https://docs.gradle.org/current/userguide/jacoco_plugin.html#sec:jacoco_report_violation_rules

File: mojio-sdk-model/build.gradle [hunk 2]
```diff
@@ -3,14 +3,13 @@ ext {
 }
 
 apply plugin: 'java'
-apply plugin: 'jacoco'
```
Comment (JesseScott) [inline mojio-sdk-model/build.gradle]: See comment above in base gradle file

File: mojio-sdk-model/build.gradle [hunk 3]
```diff
@@ -3,14 +3,13 @@ ext {
...
 apply from: project.rootProject.file('maven.gradle')
 
 sourceCompatibility = SOURCE_COMPATIBILITY
 targetCompatibility = SOURCE_COMPATIBILITY
 
 dependencies {
-    compile 'com.google.code.gson:gson:2.7'
+    compile 'com.google.code.gson:gson:2.8.5'
```
Comment (JesseScott) [inline mojio-sdk-model/build.gradle]: what's in the changelog?

File: mojio-sdk-model/src/main/java/io/moj/java/sdk/model/values/EngineOil.java [hunk 4]
```diff
@@ -11,9 +13,13 @@
     public static final String ENGINE_OIL_PRESSURE_LOW_WARNING = "EngineOilPressureLowWarning";
     public static final String ENGINE_OIL_TEMPERATURE = "EngineOilTemperature";
 
+    @SerializedName(value = "TimeStamp", alternate = {"timestamp", "Timestamp"})
     private String TimeStamp;
+    @SerializedName(value = "EngineOilLevelWarning", alternate = {"engineoillevelwarning", "engineOilLevelWarning"})
     private String EngineOilLevelWarning;
+    @SerializedName(value = "EngineOilPressureLowWarning", alternate = {"engineoilpressurelowwarning", "engineOilPressureLowWarning"})
     private Boolean EngineOilPressureLowWarning;
+    @SerializedName(value = "EngineOilTemperature", alternate = {"engineoiltemperature", "engineOilTemperature"})
```
Comment (JesseScott) [inline mojio-sdk-model/src/main/java/io/moj/java/sdk/model/values/EngineOil.java]: 👏

File: mojio-sdk-model/src/main/java/io/moj/java/sdk/model/values/EngineOilTemperature.java [hunk 5]
```diff
@@ -12,10 +14,15 @@
...
 
+    @SerializedName(value = "BaseUnit", alternate = {"baseunit", "baseUnit"})
     private String BaseUnit;
+    @SerializedName(value = "TimeStamp", alternate = {"timestamp", "Timestamp"})
     private String Timestamp;
+    @SerializedName(value = "BaseValue", alternate = {"basevalue", "baseValue"})
     private Float BaseValue;
+    @SerializedName(value = "Unit", alternate = "unit")
     private String Unit;
+    @SerializedName(value = "Value", alternate = "value")
```
Comment (JesseScott) [inline mojio-sdk-model/src/main/java/io/moj/java/sdk/model/values/EngineOilTemperature.java]: 👏

File: mojio-sdk-rest/build.gradle [hunk 6]
```diff
@@ -12,7 +11,7 @@ targetCompatibility = SOURCE_COMPATIBILITY
 dependencies {
     compile project(":mojio-sdk-model")
 
-    def retrofitVersion = '2.1.0'
+    def retrofitVersion = '2.4.0'
```
Comment (JesseScott) [inline mojio-sdk-rest/build.gradle]: what's the changelog say about this delta?

**Comment by amatkivskiy** [comment]
@dacheng81 @oleksiiborys Please check wether this PR is still relevant. 
We definitely don't need 
```
Gradle update
Removed Jacoco
```

but 
```
Updated retrofit version
Updated EngineOil and EngineOilTemperature models
```
might be needed.

**Comment by dacheng81** [comment]
Thanks @amatkivskiy . I kept neglecting the SDK PR and I didn't have the background context on this change, and it was opened a long time ago and Jesse was following this.
I'll take a closer look now.

**Pull request closed without merging**